tests/system/Database/Live/UpdateTest.php

```diff
@@ -475,14 +475,38 @@ public function testRawSqlConstraint()
 
         $builder = $this->db->table('user');
 
-        $builder->setData($data, true, 'db_myalias')
+        $builder->setData($data, true, 'myalias')
```

Comment:
Without this change, this test passes.
Is it okay?

Reply:
The alias does not matter?

When `db_myalias`:
```sql
UPDATE `db_user`
SET
`name` = `db_myalias`.`name`,
`country` = `db_myalias`.`country`
FROM (
SELECT 'Germany' `country`, 'derek@world.com' `email`, 'Derek Jones' `name`
) `db_myalias`
WHERE `db_user`.`email` = `db_myalias`.`email`
```

When `myalias`:
```sql
UPDATE `db_user`
SET
`name` = `myalias`.`name`,
`country` = `myalias`.`country`
FROM (
SELECT 'Germany' `country`, 'derek@world.com' `email`, 'Derek Jones' `name`
) `myalias`
WHERE `db_user`.`email` = `myalias`.`email`
```

Reply:
I got it.
`$this->db->protectIdentifiers('myalias.email')` converts `myalias` to `db_myalias` when `$builder->setData($data, true, 'db_myalias')`. Because `myalias` is not an alias.

This is okay. Passing `db_myalias` to `setData()` is just a misuse.